online-inference/tensorizer-isvc/gptj_6b_tensorizer/kserve/tensorizer_kserve.py

```diff
@@ -0,0 +1,92 @@
+import kserve
+import logging
+import os
+import time
+from typing import Dict
+from argparse import ArgumentParser
+from io import BytesIO
+
+import torch
+
+from tensorizer import TensorDeserializer
+from tensorizer.stream_io import open_stream
+from tensorizer.utils import no_init_or_tensor, convert_bytes, get_mem_usage
+from transformers import AutoModelForCausalLM, AutoTokenizer, AutoConfig
+
+MODEL_NAME="gptj"
+
+logging.basicConfig(level=kserve.constants.KSERVE_LOGLEVEL)
+logger = logging.getLogger(MODEL_NAME)
+logger.info(f"Model Name: {MODEL_NAME}")
+
+
+class Model(kserve.Model):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.name = name
+        self.ready = False
+        self.model_name = name
+        self.model_ref = "EleutherAI/gpt-j-6B"
+
+    def load(self):
+        logger.info(f"Loading {MODEL_NAME}")
+    
+        config = AutoConfig.from_pretrained(self.model_ref)
+
+        # This ensures that the model is not initialized.
+        with no_init_or_tensor():
+            self.model = AutoModelForCausalLM.from_config(config)
+
+        before_mem = get_mem_usage()
+
+        # Lazy load the tensors from S3 into the model.
+        start = time.time()
+        deserializer = TensorDeserializer("/mnt/pvc/gptj.tensors", plaid_mode=True)
```

Comment:
Here and in the Flask example the comment says it's loading from S3, and then proceeds to load from a PVC. Which one was intended?

Reply:
PVC